exercises/pangram/canonical-data.json

```diff
@@ -12,57 +12,81 @@
       ],
       "cases": [
         {
-          "description": "sentence empty",
+          "description": "can handle an empty sentence",
           "property": "isPangram",
           "input": "",
           "expected": false
         },
         {
-          "description": "pangram with only lower case",
+          "description": "handles an undefined message as empty message",
           "property": "isPangram",
-          "input": "the quick brown fox jumps over the lazy dog",
+          "input": "null",
+          "expected": false
+        },
+        {
+          "description": "recognizes a lower case pangram",
+          "property": "isPangram",
+          "input": "thequickbrownfoxjumpsoverthelazydog",
           "expected": true
         },
         {
-          "description": "missing character 'x'",
+          "description": "recognizes a missing character 'x'",
+          "property": "isPangram",
+          "input": "aquickmovementoftheenemywilljeopardizefivegunboats",
+          "expected": false
+        },
+        {
+          "description": "recognizes a missing character 'h'",
           "property": "isPangram",
-          "input": "a quick movement of the enemy will jeopardize five gunboats",
+          "input": "fiveboxingwizardsjumpquicklyatit",
```

Comment:
What is this testing? Why is 'h' different from 'x' and more important than 'y'?

Reply:
This is the core functionality of this exercise, so I'd argue that we should test it with more than just one possible failing input. 
The original version of the test had the following structure:
```
        {
          "description": "missing character 'x'",
          "property": "isPangram",
          "input": "a quick movement of the enemy will jeopardize five gunboats",
          "expected": false
        },
        {
          "description": "another missing character 'x'",
          "property": "isPangram",
          "input": "the quick brown fish jumps over the lazy dog",
          "expected": false
        },
```
So I guess, checking for 'h' instead of 'x' twice is already an improvement.

I'd rather like to add even more test cases, maybe about 3-5 in total. However, because the schema does not allow for multiple "expected" entries, I decided to keep it at two.

Reply:
What if it doesn't catch a missing 'a' ?
Either the 'h' test is not necessary or you're missing another 24 tests.

Reply:
Exactly! That is my problem I'm trying to balance. 
But it does not help to check for x in two cases like the original suite does atm.
If you want, then I'll do 26 test cases for every character.

But just one is simply not a reasonable option for testing the core functionality.